csharp/src/Ice/SocketStream.cs

```diff
@@ -383,6 +407,24 @@ internal virtual async ValueTask SendInitializeFrameAsync(CancellationToken canc
             // The default implementation only supports the Ice2 protocol
             Debug.Assert(_socket.Endpoint.Protocol == Protocol.Ice2);
 
+            if (frame.Size > _socket.PeerIncomingFrameMaxSize)
+            {
+                if (frame is OutgoingRequestFrame)
+                {
+                    throw new ArgumentOutOfRangeException(
```

Comment:
`ArgumentOutOfrangeException` is not the appropriate exception here. From its description:

```
An ArgumentOutOfRangeException exception is thrown when a method is invoked and at least one of the arguments passed to the method is not null and contains an invalid value that is not a member of the set of values expected for the argument. The ParamName property identifies the invalid argument, and the ActualValue property, if a value is present, identifies the invalid value.```


What about:
- make LimitExceededException a new _local_ C# exception
- add a new remote exception `ServerException` which means the server-side implementation failed, but unlike UnhandledException, may not correspond to a bug.
- update Proxy.ForwardAsync to catch the local LimitExceededException and rethrow it as a `ServerException`

Reply:
> Yes, we should. It's of course ice2-only. For ice1, we simply return 0 or -1, meaning N/A?

I don't have a preference for the value. 

For the `Connection.IdleTimeout` property, the setter throws `NotSupportedException("setting IdleTimeout is not supported with Slic")` for Slic connections. Should we consider throwing this exception here as well for consistency?

I'll fix the exceptions as suggested.

Reply:
I assume `Connection.PeerIncomingFrameMaxSize` would be get-only and throwing exceptions from getters is not a recommended pattern.